fix issue 24539, show external icon on help button

leutrimhusaj opened the pull request:
## Description
I have added the external icon (`icon={ external }`) to the Help MenuItem for edit-post and edit-site plugin. Adding as an svg was important because when i referenced it only as a string (`icon="external"`) it creates a Dashicon, which is 20x20 pixel big, it would break the padding of the text. The default icon size is 24x24 pixel and fits perfect into the more option menu.

## How has this been tested?
Open gutenberg editor for a page.
Open the "more option" menu and check/click the help button.

## Screenshots
![help-with-external-icon](https://user-images.githubusercontent.com/5269059/90518674-61fbe280-e167-11ea-8d31-d50afa6d118a.png)

## Types of changes
Bug fix for #24539 with non-breaking change

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows the accessibility standards.
- [x] My code has proper inline documentation.
- [x] I've included developer documentation if appropriate.
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

Commit:
fix issue 24539, show external icon on help button

jasmussen commented:
Thank you for the PR! This tests nicely, the code is clean and simple, and it solves the issue. The placement on the left feels a little off to me, though, and it might feel better if the icon immediately followed the link, inline in the text. We might even discuss making `external` a prop on menu items, so this gets simplified. 

However because of the simplicity of this PR, and the issue it solves, I'm not opposed to approving this as an interim solution. I also still have a PR in progress (#23552#issuecomment-656049689) which would refactor some of this stuff, so it _will_ be revisited. 

I'd love a quick sanity check, @shaunandrews or @MichaelArestad in case you have them?

Thanks again!

MichaelArestad commented:
@jasmussen It works well enough until it gets revisited. I believe I share your concern over the location of the icon. Right now it draws extra intention due to the way it aligns with other items on the list. If there's anything I can do to help your PR along, let me know.

talldan commented on packages/edit-post/src/plugins/index.js:
```diff
@@ -38,6 +39,7 @@ registerPlugin( 'edit-post', {
 							<CopyContentMenuItem />
 							<MenuItem
 								role="menuitem"
+								icon={ external }
```
Thanks so much for picking this up 👍 

In addition to the icon, the other thing that's usually present on these external links is some visually hidden text for screenreader users.

I think you should be able to follow how the External Link component does that for implementing it on these two MenuItems:
https://github.com/WordPress/gutenberg/blob/master/packages/components/src/external-link/index.js#L38-L43

Alternatively, as @jasmussen proposes a prop for the MenuItem component could be an option so that it adds the icon and the visually hidden text implicitly.

leutrimhusaj replied:
Thanks for the hint, i didn't knew about that. I have added the visually hidden text like in the example, works well.

jasmussen replied:
I'll refresh my PRs and rebase them off whatever lands here. In the mean time, this PR should not be held up. 

This is what the preview dropdown looks like:

<img width="327" alt="Screenshot 2020-08-24 at 09 50 02" src="https://user-images.githubusercontent.com/1204802/91018080-686ade00-e5ef-11ea-9b84-66c07b3f1791.png">

The same feedback applies to that, it looks a little weird on the left. But because this PR looks the same, I say let's move this forward. 👍 👍 from a design POV and thank you for the PR! If this gets a quick sanity check, we should land it.

Commit:
add visually hidden text for external link item

shaunandrews commented:
I find it a little weird, both visually (it feels unbalanced) and conceptually (the left-side space is usually reserved for the checkmark). I think it'd be better placed inline, to the right of the text, but I don't have a strong preference.

I think an ideal solution is to _not_ open an external link directly from the menu. Instead, I'd expect a help modal to appear within the editor — and from that modal, I'd find general guidance along with links to external resources.

jasmussen commented:
I have refreshed #23552, which now shows at least the preview icon on the right:

<img width="353" alt="Screenshot 2020-08-28 at 11 33 51" src="https://user-images.githubusercontent.com/1204802/91546237-6de36380-e922-11ea-9c1e-d54edce668d9.png">

If that PR lands first, rebasing this one should address the most pressing feedback about the icon positioning. That does not preclude a further PR that changes that link to open a modal dialog instead — totally fair feedback — but until that, this one would be a nice little improvement.

shaunandrews commented:
👍

Commit:
unite duplicate import

leutrimhusaj commented:
I have rebased the PR and fixed the conflicts. This PR looks like this now.

<img width="435" alt="help-with-external-icon-right" src="https://user-images.githubusercontent.com/5269059/92380831-15724a00-f10a-11ea-8e91-2fc981b05766.png">

@jasmussen @talldan Is there anything else to do for this minimal PR?

The other feature should be done in other PR in my opinion.

jasmussen reviewed (approved):
This seems like a good small win. Thanks for the PR!

Before:

<img width="425" alt="Screenshot 2020-09-08 at 11 11 08" src="https://user-images.githubusercontent.com/1204802/92457742-e53bb180-f1c4-11ea-84a0-b3d69e5e6578.png">

After:

<img width="441" alt="Screenshot 2020-09-08 at 11 15 23" src="https://user-images.githubusercontent.com/1204802/92457752-e836a200-f1c4-11ea-9c39-84bc97697a3f.png">

Markup:

<img width="547" alt="Screenshot 2020-09-08 at 11 15 33" src="https://user-images.githubusercontent.com/1204802/92457763-ebca2900-f1c4-11ea-8bf9-7945eed2b93d.png">
